src/test/java/org/folio/rest/api/DereferencedItemStorageTest.java

```diff
@@ -0,0 +1,245 @@
+package org.folio.rest.api;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.support.ResponseHandler.json;
+import static org.folio.rest.support.http.InterfaceUrls.holdingsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.instancesStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.itemsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.dereferencedItemStorage;
+import static org.folio.util.StringUtil.urlEncode;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.jaxrs.model.DereferencedItem;
+import org.folio.rest.jaxrs.model.DereferencedItems;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+
+
+@RunWith(JUnitParamsRunner.class)
+public class DereferencedItemStorageTest extends TestBaseWithInventoryUtil {
+  private static final UUID smallAngryPlanetId = UUID.randomUUID();
+
+  // see also @BeforeClass TestBaseWithInventoryUtil.beforeAny()
+
+  @BeforeClass
+  public static void beforeTests() throws InterruptedException, ExecutionException, TimeoutException {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    JsonObject smallAngryPlanet = smallAngryPlanet(smallAngryPlanetId, holdingsRecordId);
+    JsonObject nod = nod(UUID.randomUUID(), holdingsRecordId);
+    JsonObject uprooted = uprooted(UUID.randomUUID(), holdingsRecordId);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
```

Comment:
The request isn't a get, so I think this might be inappropriately named.
```suggestion
    CompletableFuture<Response> postCompleted = new CompletableFuture<>();
```

```suggestion
    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
```